gtsfm/bundle/bundle_adjustment.py

```diff
@@ -36,7 +38,7 @@ class BundleAdjustmentOptimizer(NamedTuple):
 
     output_reproj_error_thresh: float
 
-    def run(self, initial_data: GtsfmData) -> GtsfmData:
+    def run(self, initial_data: GtsfmData) -> Tuple[GtsfmData, GtsfmMetricsGroup]:
```

Comment:
is there an advantage to returning the metric group here, instead of just saving it to disk right away, and forgetting about it?

Reply:
Yes, for plotting the metrics.

Reply:
We’re just loading them later from JSON, though, right?

Reply:
No, we are also saving the plots when the pipeline runs. Refer to scene_optimizer.py. 
I have just added a script for convenience to plot them from JSON, but I think these should be plotted when the pipeline runs. That is more programmatic and less prone to errors.

Reply:
Agree with Akshay.

Reply:
okay, i see now